### PR TITLE
expanded verbose for all algorithms, made verbose a boolean for all algorithms

### PR DESCRIPTION
**Pull request by vroulet**
Replaced "Error" by specific name of error considered and added other information when verbose=True.
Should be useful for debugging more quickly.
All verbose are now boolean too (some were int).

### CODE REVIEW

File: jaxopt/_src/armijo_sgd.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ class ArmijoSGD(base.StochasticSolver):
   maxiter: int = 500
   maxls: int = 15
   tol: float = 1e-3
-  verbose: int = 0
+  verbose: bool = False
```
Comment (mblondel) [inline jaxopt/_src/armijo_sgd.py:227]: The idea of using an `int` was that we could have several levels of verbosity:
- 1: verbose
- 2: very verbose
- 3: very very verbose

For example, 2 could be used to print linesearch information.

Do you think it's unnecessary?

File: jaxopt/_src/armijo_sgd.py [hunk 2]
```diff
@@ -315,6 +315,15 @@ def update(self, params, state, *args, **kwargs) -> base.OptStep:
                              stepsize=jnp.asarray(stepsize, dtype=dtype),
                              velocity=next_velocity)
 
+    if self.verbose:
+      self.log_info(
+          next_state,
+          error_name="Grad. Norm",
+          additional_info={
+              'Obj. Val.': next_state.value,
```
Comment (mblondel) [inline jaxopt/_src/armijo_sgd.py:323]: I would avoid abbreviations, e.g., I prefer `Objective value` here.

Comment (vroulet) [reply jaxopt/_src/armijo_sgd.py]: Done

File: jaxopt/_src/armijo_sgd.py (hunk 1 above)
Comment (vroulet) [reply jaxopt/_src/armijo_sgd.py]: I did a two level verbose for the solvers with linesearch as you suggested. 
I made all verbose types to be boolean or int to be consistent.
Thanks for the suggestion!

**Review by mblondel — APPROVED** [review]
Thanks Vincent

**Pull request merged**